lib/SILOptimizer/Mandatory/OSLogOptimization.cpp

```diff
@@ -482,14 +489,13 @@ destroyFoldedValueAtEndOfUse(SILValue foldedVal, SILValue originalVal,
   }
   assert(originalVal.getOwnershipKind() == ValueOwnershipKind::Guaranteed);
 
-  // Here, the original value may be at +0 and hence may not be released.
-  // However, the folded value should always be released.
-  SmallVector<SILInstruction *, 2> lifeTimeEndInstsOfOriginal;
-  getLifetimeEndInstructionsOfSILValue(originalVal, lifeTimeEndInstsOfOriginal);
-
-  if (!lifeTimeEndInstsOfOriginal.empty()) {
-    // Here, the original value is released, and so would be the folded value.
-    return;
+  // Here, the original value may be at +0 or + 1. In the latter case, the
+  // original value will be released and so would be the folded value.
+  for (Operand *use : originalVal->getUses()) {
```

Comment:
In OSSA, you could just check that the ownership kind is owned, i.e.:

```
originalVal->getOwnershipKind() == ValueOwnershipKind::Owned
```

Reply:
>  what about a SIL test case?

Sure. I will add a SIL test suite for the OSLogOptimization pass.